api/src/chat/chatRoom.dto.ts

```diff
@@ -31,8 +37,55 @@ export interface MessageInterface {
 	updated?: Date;
 	game?: boolean;
 	sender_avatar?: string;
+	type: string;
+	cutomMessageData?: {href: string, text: string};
 }
 
+export interface Rooms {
+	id: number;
+	name: string;
+	owner: number;
+	admins: number[];
+	banned?: number[];
+	muted?: Record<number, Date>;
+	users: number[];
+	status: string; //public, private
+	password: boolean; //true or false?
+	created?: Date;
+	updated?: Date;
+	messages?: MessageInterface[];
+}
+
+export interface ErrorMessage{
+	msg: string;
+	status_code: number;
+	room?: string;
+}
+
+export interface createRoomDto{
```

Comment:
These DTOs need to be validated better. Right now you can create a room with the name "~~~", but sending a message to it will fail. This is because the message thinks the room name isn't valid.
Also more thought needs to be put into what is and isn't valid. For example, right now it isn't possible to send messages with spaces in it.